keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/AdminService.kt

```diff
@@ -66,37 +65,38 @@ class AdminService(
   }
 
   /**
-   * updates old artifact version records with their metadata, if exists, by type [deb/docker/npm]
+   * Updates artifact version records with the corresponding metadata, if available, by type [deb/docker/npm]
    */
-  fun backfillOldArtifactVersions(type: String) {
+  fun backFillArtifactMetadata(type: String) {
     val artifactSupplier = artifactSuppliers.supporting(type)
-    log.debug("starting to backfill old artifacts versions with artifact metadata")
-    //1. get all registered artifacts
+    log.debug("Starting to back-fill old artifacts versions with artifact metadata...")
+    // 1. get all registered artifacts
     val deliveryArtifacts = repository.getAllArtifacts(type)
-    //2. for each artifact, fetch all versions
+    // 2. for each artifact, fetch all versions
     deliveryArtifacts.forEach { artifact ->
       val versions = repository.artifactVersions(artifact)
       versions.forEach { version ->
+        sleep(5000) // sleep a little in-between versions to cut the instance where this runs some slack
```

Comment:
Added a little delay here to allow some breathing room for the instance where this runs...